packages/aws-cdk-lib/aws-ecr/lib/repository.ts

```diff
@@ -373,17 +379,37 @@ export abstract class RepositoryBase extends Resource implements IRepository {
     return ret;
   }
 
+  /**
+   * Grant the given identity permissions to use the images in this repository
+   */
+  public grantPush(grantee: iam.IGrantable) {
+    const ret = this.grant(grantee, ...perm.REPO_PUSH_ACTIONS);
+    iam.Grant.addToPrincipal({
+      grantee,
+      actions: ['ecr:GetAuthorizationToken'],
+      resourceArns: ['*'],
+      scope: this,
+    });
+
+    return ret;
+  }
+
   /**
    * Grant the given identity permissions to pull and push images to this repository.
    */
   public grantPullPush(grantee: iam.IGrantable) {
-    this.grantPull(grantee);
-    return this.grant(grantee,
-      'ecr:PutImage',
-      'ecr:InitiateLayerUpload',
-      'ecr:UploadLayerPart',
-      'ecr:CompleteLayerUpload',
+    const ret = this.grant(grantee,
+      ...perm.REPO_PULL_ACTIONS,
+      ...perm.REPO_PUSH_ACTIONS,
     );
+    iam.Grant.addToPrincipal({
```

Comment:
To deal with the duplication issue, you can declare both sets of actions as `Set<string>`, both of which containing `'ecr:GetAuthorizationToken'`. Then you can transform them to arrays before passing them to `this.grant()`.

Reply:
Thank you for the suggestion. Unfortunately, the ecr:GetAuthorizationToken action is invoked not against a registry, but against the entire ECR service, requiring its resource to be set as '*'. Therefore, I believe it's difficult to handle it along with other Actions within the grant. With this in mind, if there are any necessary refactorings, I would appreciate your guidance.